### PR TITLE
Fix icons and modals in 0.8, support 0.8 in the angular explorer app and add missing 0.8 integration tests

### PR DESCRIPTION
**Pull request by josemontespg**
This PR Fixes the modal and icon components in the 0.8 basic catalog. Also implements a dropdown in the Angular explorer sample app to switch between v0.8 and v0.9 specifications and renderers, to help with manual and automated testing.

## Key changes:
*   **Modal**: Fixed modal overlay rendering inline, instead of in an overlay.
*   **Icon**: Fixed icons not rendering when the icon name used camelCase or TitleCase.
*   **Version Toggle**: Added a dropdown in the sidebar to switch between 0.8 and 0.9.
*   **URL Parameter**: Added support for `?version=0.8` URL parameter to load a specific version directly.
*   **Examples**: Updated `generate-examples.mjs` to bundle examples for both versions.
*   **Integration tests**: Added integration tests for the Angular explorer app to test the 0.8 core, renderer and basic catalog. These tests run also in CI.

## Why:
The motivation for these tests is http://b/511068435#comment3 because we were missing a test for the 0.8 modal component. I found that we were missing 0.8 tests for the explorer app, which is already a great sample app to exercise many different features and edge cases for 0.8.

## Testing:
*   Verified the 0.8 examples by running the Angular explorer app. The URL param works, and components/icons render correctly.
*   Added unit tests for the icon and modal fixes. Also, added one integration test for each of the example JSONs

### CODE REVIEW

File: renderers/angular/src/v0_8/components/icon.ts [hunk 1]
```diff
@@ -45,5 +45,9 @@ import type {IconNode, StringValue} from '../types';
 })
 export class Icon extends DynamicComponent<IconNode> {
   readonly name = input<StringValue | null>(null);
-  protected readonly resolvedName = computed(() => this.resolvePrimitive(this.name()));
+  protected readonly resolvedName = computed(() => {
+    const rawName = this.resolvePrimitive(this.name());
+    if (!rawName) return '';
+    return rawName.replace(/[A-Z]/g, letter => `_${letter.toLowerCase()}`);
```
[the remark this answers is not in the document]
Comment (josemontespg) [reply renderers/angular/src/v0_8/components/icon.ts]: Done

**Comment by polina-c** [comment]
For your future PRs: try to separate code reorg and code changes (make two different PRs), to make it easier to make reviews.

**Comment by polina-c** [comment]
Would you mind to link an issue that it contributes to, in the description?

(if you did not know that PR can auto-resolve an issue: https://github.com/flutter/genui/pull/914)

**Comment by josemontespg** [comment]
> For your future PRs: try to separate code reorg and code changes (make two different PRs), to make it easier to make reviews.

Sorry for the large change, I'll keep this in mind in the future. Thank you for taking the time to review it!


> Would you mind to link an issue that it contributes to, in the description?

Added a link to the internal issue in the description that motivates this PR

**Review by polina-c — APPROVED** [review]
LGTM, but I would want also review from someone who is more familiar with our ts codebase.

File: renderers/angular/a2ui_explorer/src/app/tests/test_utils.ts [hunk 2]
```diff
@@ -33,9 +33,24 @@ export async function loadExample(exampleName: string) {
...
+  const examples = version === '0.9' ? EXAMPLES_V09 : EXAMPLES_V08;
+  let example = examples.find(ex => ex.name === exampleName);
+
+  if (version === '0.8') {
+    if (!example) {
+      example = examples.find(ex => ex.name === `${exampleName} (basic)`);
+    }
+    if (!example) {
+      example = examples.find(ex => ex.name === `${exampleName} (minimal)`);
+    }
```
Comment (ditman) [inline renderers/angular/a2ui_explorer/src/app/tests/test_utils.ts:45]: I think we should remove the "minimal" catalog and examples. They don't add much value once the renderer is fully built (instead, they add maintenance burden like here)

File: renderers/angular/a2ui_explorer/src/app/agent-stub.service.ts [hunk 3]
```diff
@@ -37,39 +37,24 @@ interface SubmitFormContext {
   name?: string;
 }
 
-/**
- * A stub service that simulates an A2UI agent.
- * It listens for actions and responds with data model updates or new surfaces.
- */
```
Comment (ditman) [inline renderers/angular/a2ui_explorer/src/app/agent-stub.service.ts:43]: TSDoc was useful, why would we want to remove this?

File: renderers/angular/a2ui_explorer/src/app/demo.component.ts [hunk 4]
```diff
@@ -61,8 +72,9 @@ import {ActionDispatcher} from './action-dispatcher.service';
           <p class="subtitle">{{ selectedExample.description }}</p>
         </div>
         <div class="canvas-frame">
-          <div *ngIf="surfaceId" class="rendered-content">
-            <a2ui-v09-surface [surfaceId]="surfaceId"> </a2ui-v09-surface>
+          <div *ngIf="surfaceId" class="rendered-content" [attr.data-version]="version">
+            <a2ui-v09-surface *ngIf="version === '0.9'" [surfaceId]="surfaceId"> </a2ui-v09-surface>
```
Comment (ditman) [inline renderers/angular/a2ui_explorer/src/app/demo.component.ts]: Remove the whtiespace inside the `a2ui-v09-surface`?

File: renderers/angular/a2ui_explorer/src/app/demo.component.ts [hunk 5]
```diff
@@ -511,10 +564,14 @@ import {ActionDispatcher} from './action-dispatcher.service';
...
   private agentStub = inject(AgentStubService);
+  private actionDispatcher = inject(ActionDispatcher);
   private cdr = inject(ChangeDetectorRef);
+  private messageProcessorV08 = inject(MessageProcessorV08);
+  private themeV08 = inject(ThemeV08);
 
-  examples = EXAMPLES;
-  selectedExample: Example | undefined = undefined;
+  version: '0.8' | '0.9' = '0.9';
+  examples: Array<Example | Example_08> = EXAMPLES_V09;
```
Comment (ditman) [inline renderers/angular/a2ui_explorer/src/app/demo.component.ts:573]: Would it make sense to have DemoComponentV08 and DemoComponentV09 where each one of them knows how to handle its version of the protocol (with duplicated code) instead of adding many "if version == 0.8 else..." here?

**Review by ditman — COMMENTED** [review]
I am hesitant of adding v0.8 "inline" to the explorer app; I think this should be a slightly different "demo" component.

Also, needs CHANGELOG updates.

Also also, @polina-c is right, can we please split this PR in at least 3 or 4 (support angular explorer app 0.8, add missing tests, fix icons, fix modals?)

File: renderers/angular/a2ui_explorer/src/app/demo.component.ts (hunk 4 above)
Comment (josemontespg) [reply renderers/angular/a2ui_explorer/src/app/demo.component.ts]: Done

File: renderers/angular/a2ui_explorer/src/app/agent-stub.service.ts (hunk 3 above)
Comment (josemontespg) [reply renderers/angular/a2ui_explorer/src/app/agent-stub.service.ts:43]: Done

File: renderers/angular/a2ui_explorer/src/app/tests/test_utils.ts (hunk 2 above)
Comment (josemontespg) [reply renderers/angular/a2ui_explorer/src/app/tests/test_utils.ts:45]: Sounds good to me. Is there alignment on doing this? Created #1432 to track

**Comment by josemontespg** [comment]
Thanks for the feedback! I'm closing this PR in favor of smaller PRs
